src/hotspot/share/prims/forte.cpp

```diff
@@ -522,12 +522,14 @@ static void forte_fill_call_trace_given_top(JavaThread* thd,
 extern "C" {
 JNIEXPORT
 void AsyncGetCallTrace(ASGCT_CallTrace *trace, jint depth, void* ucontext) {
-  JavaThread* thread;
 
-  if (trace->env_id == NULL ||
-    (thread = JavaThread::thread_from_jni_environment(trace->env_id)) == NULL ||
-    thread->is_exiting()) {
+  // Can't use thread_from_jni_environment as it may also perform a VM exit check that is unsafe to
+  // do from this context.
+  Thread* raw_thread = Thread::current_or_null_safe();
+  JavaThread* thread;
 
+  if (trace->env_id == NULL || raw_thread == NULL || !raw_thread->is_Java_thread() ||
+     (thread = ((JavaThread*)raw_thread))->is_exiting()) {
```

Comment:
One minor diff: indentation

Reply:
What is the problem with this indentation?

Reply:
The original change indents line 532 by four, the backport only by three.

Reply:
Thanks, I'll change it accordingly.